src/transformers/training_args.py

```diff
@@ -963,6 +964,19 @@ class TrainingArguments:
             )
         },
     )
+    timeout: Optional[int] = field(
+        default=1800,
+        metadata={
+            "help": (
+                "Overrides the default timeout defined by PyTorch and"
+                " introduces a way to prevent Socket Timeout when mapping large datasets."
+                " Expects timeout in seconds. Used for timeout argument in"
+                " torch.distributed.init_process_group calls. Please refer the PyTorch documentation"
+                " https://pytorch.org/docs/stable/distributed.html#torch.distributed.init_process_group"
+                " for more information."
```

Comment:
This is a bit too long here, and the docstring for this new argument is missing. I'd limit the help here to
```suggestion
                "Overrides the default timeout for distributed training (value should be given in seconds).
```"
and you can add more info in the docstring.

Reply:
Sounds perfect! Thanks @sgugger. I will push the changes in a couple of minutes.